frontend/src/components/tables/DocumentDirectoryTable.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { useState } from "react";
+import {
+  Button,
+  Checkbox,
+  Table,
+  Thead,
+  Tbody,
+  Tr,
+  Th,
+  Td,
+  TableContainer,
+} from "@chakra-ui/react";
+
+// TODO: remove once connected to backend
+const dummyData = [
+  {
+    id: "1",
+    name: "A",
+    created: "2024-03-01",
+    type: "Form",
+    comments: "blah blah",
+  },
+  {
+    id: "2",
+    name: "B",
+    created: "2024-03-02",
+    type: "Form",
+    comments: "blah blah",
+  },
+  {
+    id: "3",
+    name: "C",
+    created: "2024-03-03",
+    type: "Form",
+    comments: "blah blah",
+  },
+];
+
+const DocumentDirectoryTable = () => {
+  const documents = dummyData; // TODO: replace with call to backend
+
+  const [selectedDocumentIds, setSelectedDocumentIds] = useState<Set<string>>(
+    new Set(),
+  );
+
+  const changeRowCheckbox = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const selectedId = event.target.id;
+    const isChecked = event.target.checked;
+    if (isChecked) {
+      selectedDocumentIds.add(selectedId);
+    } else {
+      selectedDocumentIds.delete(selectedId);
+    }
+    setSelectedDocumentIds((prev) => new Set(prev));
```

Comment:
Doing `Set.add` and `Set.delete` won't trigger a React state update, because the object itself has not changed (even though its contents did).
To force a state update, I'm creating a new Set copied off the mutated previous Set.